src/test/java/io/wisetime/connector/jira/database/JiraDbTest.java

```diff
@@ -4,39 +4,310 @@
+    saveProject(projectId, issue.getProjectKey());
+    saveJiraIssue(projectId, issue);
+
+    jiraDb.updateIssueTimeSpent(issue.getId(), 700);
+
+    assertThat(jiraDb.findIssueByTagName(issue.getProjectKey() + "-" + issue.getIssueNumber()).get().getTimeSpent())
+        .as("should be able to update total time spent")
+        .isEqualTo(700);
+  }
+
+  @Test
+  void createWorklog_newRecord() {
+    ZoneId sydneyTz = ZoneId.of("Australia/Sydney");
+    ZoneId perthTz = ZoneId.of("Australia/Perth");
+
+    // specify timezone to use
+    saveDefaultTimeZone(1, perthTz.getId());
+
+    Worklog workLogWithSydneyTz = FAKE_ENTITIES.randomWorklog(sydneyTz);
+    Worklog workLogWithPerthTz = ImmutableWorklog.builder().from(workLogWithSydneyTz)
+        .created(ZonedDateTime.of(workLogWithSydneyTz.getCreated(), ZoneOffset.UTC).toLocalDateTime().withNano(0))
+        .build();
+
+    Optional<Long> startingWorklogId = jiraDb.getWorklogSeqId();
+    assertThat(startingWorklogId).isEmpty();
+
+    jiraDb.createWorklog(workLogWithSydneyTz);
+
+    assertThat(getWorklog(10299).get()) // 10299 is the starting work log seq id we set if table is empty
+        .as("work log should be saved with created time set the zone specified")
+        .isEqualTo(workLogWithPerthTz);
+  }
+
+  @Test
+  void createWorklog_withExistingWorklog() {
+    ZoneId sydneyTz = ZoneId.of("Australia/Sydney");
+    ZoneId perthTz = ZoneId.of("Australia/Perth");
+
+    // specify timezone to use
+    saveDefaultTimeZone(1, perthTz.getId());
+
+    // create work log
+    Worklog workLog1WithSydneyTz = FAKE_ENTITIES.randomWorklog(sydneyTz);
+    jiraDb.createWorklog(workLog1WithSydneyTz);
+
+    Worklog workLog2WithSydneyTz = FAKE_ENTITIES.randomWorklog(sydneyTz);
+    Worklog workLog2WithPerthTz = ImmutableWorklog.builder().from(workLog2WithSydneyTz)
+        .created(ZonedDateTime.of(workLog2WithSydneyTz.getCreated(), perthTz).toLocalDateTime().withNano(0))
+        .build();
+    Optional<Long> currentWorkLogId = jiraDb.getWorklogSeqId();
+    assertThat(currentWorkLogId)
+        .as("should contain the work log id of the previously created work log")
+        .isPresent();
+
+    jiraDb.createWorklog(workLog2WithSydneyTz);
+
+    assertThat(getWorklog(currentWorkLogId.get() + 199).get()) // we increment 199 to generate new work log seq id
+        .as("work log should be saved with created time set the zone specified")
+        .isEqualTo(workLog2WithPerthTz);
+  }
+
+  private void saveProject(Long projecId, String projectKey) {
+    query
+        .update(
+            String.format("INSERT INTO project (id, pkey) VALUES (%d, '%s')", projecId, projectKey)
+        )
+        .run();
+  }
+
+  private void saveJiraIssue(Long projectId, Issue issue) {
+    query
+        .update(
+            String.format(
+                "INSERT INTO jiraissue (id, project, issuenum, summary, timespent) VALUES (%d, %d, '%s', '%s', %d)",
```

Comment:
Fluentjdbc.query() has tools to handle query with params (see comment above). It is preferable over String.format() as it has correct escaping (e.g. String.format will fail if `issue.summary` will have single quote in it, etc).